File: code/PokemonSystem.BillsPC/Application/ViewModel/PokemonDetail.cs
using PokemonSystem.Common.ValueObjects;

namespace PokemonSystem.BillsPC.Application.ViewModel
{
    public class PokemonDetail
    {
        public Guid Id { get; protected set; }
        public string? Nickname { get; protected set; }
        public string Gender { get; protected set; } = string.Empty;
        public int SpeciesId { get; protected set; }
        public string SpeciesName { get; protected set; } = string.Empty;
        public short Type1 { get; protected set; }
        public short? Type2 { get; protected set; }
        public uint Level { get; protected set; }
        public List<Move> LearntMoves { get; protected set; } = new List<Move>();
        public Stats Stats { get; protected set; } = new Stats(0, 0, 0, 0, 0, 0);
    }
}

File: code/front-app/src/models/PokemonDetail.ts
import { Move } from "./Move";
import { PokemonType } from "./PokemonType";
import { Stats } from "./Stats";

export class PokemonDetail {
    id: string;
    nickname: string | null;
    gender: string;
    speciesId: number;
    speciesName: string;
    type1: PokemonType;
    type2: PokemonType;
    level: number;
    learntMoves: Move[];
    stats: Stats;    
}